pointcloud/kdtree.go

```diff
@@ -87,7 +88,7 @@ func (kdv kdValuesSlicer) Swap(i, j int) {
 
 // KDTree extends PointCloud and orders the points in 3D space to implement nearest neighbor algos.
 type KDTree struct {
-	tree   *kdtree.Tree
+	Tree   *kdtree.Tree
```

Comment:
Had to make this field public and create some unsafe functions because of the way I'm preemptively splitting up the tree and trying to run the clustering algorithm on the children at level 2.